mmv1/templates/terraform/pre_update/alloydb_cluster.go.erb

```diff
@@ -1,3 +1,38 @@
+// Implementation for cluster upgrade
+if d.HasChange("database_version") && !tpgresource.IsEmptyValue(reflect.ValueOf(d.Get("database_version"))) {
+    upgradeUrl := strings.Split(url, "?updateMask")[0] + ":upgrade"
+    patchObj := make(map[string]interface{})
+    patchObj["version"] = obj["databaseVersion"]
+    if bp, err := tpgresource.GetBillingProject(d, config); err == nil {
+        billingProject = bp
+    }
+    upgradeUrl = strings.Replace(upgradeUrl, "/v1/", "/v1alpha/", 1)
```

Comment:
I wonder the reason to replace `/v1/` with `/v1alpha/` is this feature is unavailable in v1? 

If the feature is only available in beta but not in v1, a version guard is needed for the code to upgrade  `database_version`. Example: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/decoders/containeranalysis_occurrence.go.erb#L15

Reply:
Yes. At the time I raised the PR, the API rollout was underway which is almost complete now and hence I have removed this line which replaces `v1` with `v1alpha`.

This should work with all the API versions now.